{python3packages.py,}meshlab: 2023.12 -> 2025.07 + use system dependencies , and update dependencies

nim65s opened the pull request:
fix #451139

Hi,

This PR update meshlab & python3Packages.pymeshlab from 2023.12 to 2025.07, among other things to fix build with CMake v4.

It also update their dependencies (levmar, lib3mf, tinygltf, vcg, corto, and nexus) and their packaging to use standard CMake exports (ref. https://github.com/cnr-isti-vclab/vcglib/pull/248)

While here, meshlab-unstable is dropped, since it was older than 2025.07, cc. @pca006132

When I started to work on this, I did not see the work from #396295 (merged only 4 days ago), so I tried to merge this work with mine, but a few things might seem odd in the history. cc. @yzx9

I tested that by opening meshlab GUI and loading a mesh (only on linux) and trying a simple mesh decimation from python (linux & darwin):
```python
$ nix-shell -I nixpkgs=. -p 'python3.withPackages(p: [p.pymeshlab])' --command python
>>> from pymeshlab import MeshSet
>>> ms = MeshSet()
>>> ms.load_new_mesh("/…/base_link.STL")
>>> ms.meshing_decimation_quadric_edge_collapse()
>>> ms.save_current_mesh("/…/base_link2.STL")
```

And, yes, `base_link2.STL` takes less disk space than `base_link.STL` (a random file from example-robot-data)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nim65s committed:
meshlab: adopt

nim65s committed:
meshlab: 2023.12 -> 2025.07 + use system dependencies

nim65s committed:
levmar: install new header

lm.h only contains a deprecation warning now

nim65s committed:
corto: 0-unstable-2024-04-05 -> v2025.07

nim65s committed:
lib3mf: adopt

nim65s committed:
lib3mf: fix CMake export

nim65s committed:
levmar: adopt

nim65s committed:
vclab-nexus: init at 2025.05

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

nim65s committed:
vcg: adopt

nim65s committed:
vcg: clean

nim65s committed:
vcg: switch to CMake

nim65s committed:
python3Packages.pymeshlab: 2023.12 -> 2025.7 + use system dependencies

nim65s committed:
tinygltf: init at 2.9.6

nim65s committed:
meshlab-unstable: drop

nim65s committed:
libigl: dont install opengl headers

fix meshlab on darwin:
```
[ 47%] Building CXX object src/meshlabplugins/filter_cubization/CMakeFiles/filter_cubization.dir/filter_cubization_autogen/mocs_compilation.cpp.o
In file included from /tmp/nix-build-meshlab-2025.07.drv-0/source/build/src/meshlabplugins/filter_cubization/filter_cubization_autogen/mocs_compilation.cpp:2:
In file included from /tmp/nix-build-meshlab-2025.07.drv-0/source/build/src/meshlabplugins/filter_cubization/filter_cubization_autogen/EWIEGA46WW/moc_filter_cubization.cpp:10:
In file included from /tmp/nix-build-meshlab-2025.07.drv-0/source/build/src/meshlabplugins/filter_cubization/filter_cubization_autogen/EWIEGA46WW/../../../../../../src/meshlabplugins/filter_cubization/filter_cubization.h:28:
In file included from /tmp/nix-build-meshlab-2025.07.drv-0/source/src/common/../common/plugins/interfaces/filter_plugin.h:29:
In file included from /tmp/nix-build-meshlab-2025.07.drv-0/source/src/common/../common/plugins/interfaces/../../ml_document/mesh_document.h:27:
In file included from /tmp/nix-build-meshlab-2025.07.drv-0/source/src/common/../common/plugins/interfaces/../../ml_document/mesh_model.h:26:
In file included from /nix/store/18nlfi993h20y6dd1kfg8zm9gaglwygn-glew-2.2.0-dev/include/GL/glew.h:1217:
In file included from /nix/store/rhlyld20zdzdi5wghb4b12gw52s9whr9-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Headers/glu.h:9:
/nix/store/zchkzj4md6q4lq09iidcm4c8n7r22k46-libigl-2.6.0/include/igl/OpenGL/gl.h:23:10: fatal error: 'glad/glad.h' file not found
   23 | #include <glad/glad.h>
      |          ^~~~~~~~~~~~~
```

nim65s committed:
python3Packages.pymeshlab: fix on darwin

nim65s committed:
meshlab: darwin updates

- fix CMake exports
- add missing library
- update to llvm 19

Co-authored-by: Zexin Yuan <gh@yzx9.xyz>

pca006132 commented on pkgs/by-name/me/meshlab/1617_cmake-use-system-dependencies-install-exports.patch:
Have you considered making an upstream PR for this?

nim65s replied:
Yes, the upstream PR is here: https://github.com/cnr-isti-vclab/meshlab/pull/1617

nim65s commented on pkgs/top-level/all-packages.nix:
```diff
@@ -12184,12 +12184,6 @@ with pkgs;
 
   meshlab = callPackage ../by-name/me/meshlab/package.nix {
     stdenv = if stdenv.hostPlatform.isDarwin then llvmPackages_18.stdenv else stdenv;
```
@yzx9: maybe we can remove that too, right ? And move meshlab to pkgs/by-name ?

yzx9 replied:
Yes, we should able to remove the entry from `all-packages.nix`

nim65s committed:
meshlab: remove from all-packages, use by-name

SuperSandro2000 reviewed (commented):
otherwise LGTM

nim65s committed:
lib3mf: replace-warn -> replace-fail

Co-authored-by: Sandro <7258858+SuperSandro2000@users.noreply.github.com>

nim65s commented on pkgs/by-name/li/libigl/package.nix:
```diff
@@ -20,6 +20,7 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   installPhase = ''
     mkdir -p $out/include
     cp -r include/igl $out/include
+    rm -rf $out/include/igl/opengl
```
Yes, those are installed on all platforms : `installPhase` is `cp -r include/igl $out/include`

nim65s commented on pkgs/by-name/vc/vclab-nexus/package.nix:
```diff
@@ -0,0 +1,70 @@
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "vclab-nexus"; # nexus is already taken
```
I hadn't thought of it, thanks ! :)

nim65s committed:
meshlab: darwin plist: substitute -> patch

Co-authored-by: Sandro <7258858+SuperSandro2000@users.noreply.github.com>

nim65s committed:
python3Packages.pymeshlab: dont propagate hook

Co-authored-by: Sandro <7258858+SuperSandro2000@users.noreply.github.com>

nim65s commented on pkgs/by-name/li/lib3mf/package.nix:
```diff
@@ -93,6 +93,11 @@ stdenv.mkDerivation (finalAttrs: {
       --replace-warn "EVP_cleanup();" "" \
       --replace-warn "CRYPTO_cleanup_all_ex_data();" ""
+
+    # Fix CMake export
+    substituteInPlace cmake/lib3mfConfig.cmake \
+      --replace-fail "$""{LIB3MF_ROOT_DIR}/include" "$""{LIB3MF_ROOT_DIR}/include/lib3mf" \
+      --replace-fail "$""{LIB3MF_ROOT_DIR}/lib" "$out/lib"
```
https://github.com/3MFConsortium/lib3mf/pull/434

nim65s commented:
Thanks for the reviews ! 
I've incorporated that, and rebased on the current master, after a merge from staging containing the most urgent CMake v4 parts.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 445078`
Commit: `3293b64dca95a2a3b12d651b6de43d29b7f5fc1f`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 21 packages built:</summary>
  <ul>
    <li>corto</li>
    <li>kicadAddons.kikit</li>
    <li>kicadAddons.kikit-library</li>
    <li>kikit</li>
    <li>kikit.dist</li>
    <li>levmar</li>
    <li>lib3mf</li>
    <li>lib3mf.dev</li>
    <li>libigl</li>
    <li>meshlab</li>
    <li>openbrf</li>
    <li>openmvs</li>
    <li>openscad</li>
    <li>openscad-unstable</li>
    <li>python312Packages.pymeshlab (python313Packages.pymeshlab)</li>
    <li>snis-unwrapped</li>
    <li>tinygltf</li>
    <li>vcg</li>
    <li>vclab-nexus</li>
    <li>vclab-nexus.bin</li>
    <li>wings</li>
  </ul>
</details>

### `aarch64-darwin`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>meshlab</li>
    <li>openscad</li>
    <li>python312Packages.pymeshlab (python313Packages.pymeshlab)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>corto</li>
    <li>levmar</li>
    <li>lib3mf</li>
    <li>lib3mf.dev</li>
    <li>libigl</li>
    <li>openmvs</li>
    <li>openscad-unstable</li>
    <li>tinygltf</li>
    <li>vcg</li>
    <li>vclab-nexus</li>
    <li>vclab-nexus.bin</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>openscad</summary>
<pre>  294 | #  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
      |                                 ^
/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtCore/qcompilerdetection.h:691:55: note: expanded from macro &#x27;Q_DECL_DEPRECATED_X&#x27;
  691 | #    define Q_DECL_DEPRECATED_X(text) __attribute__ ((__deprecated__(text)))
      |                                                       ^
/nix/store/l58v977vq4qy58m1bw60bzrpkywizz2b-clang-wrapper-21.1.1/bin/clang++ -c -pipe -stdlib=libc++ -DSTACKSIZE=8388608 -O2  -arch arm64 -isysroot /nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -mmacosx-version-min=10.9 -I/nix/store/2wsfr9fm6xzy0nrmisxdb01si4d1lyd4-cairo-1.18.4-dev/include/cairo -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include -I/nix/store/7qkcsn97gyn3xnd5jsnrm8mib62pa81c-libzip-1.11.4-dev/../7qkcsn97gyn3xnd5jsnrm8mib62pa81c-libzip-1.11.4-dev/include -I/nix/store/3ydndym6ivg31py7l4flz3lpc28ww15l-libxml2-2.14.6-dev/include/libxml2 -I/nix/store/fl9rvp5x46ahiw9s7inzl5w3bzjjdcfl-lib3mf-2.4.1-dev/include/lib3mf/Bindings/Cpp -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/jxw17p0zgbsl58qn711g6rsf8wgliclz-fontconfig-2.17.1-dev/include -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/w8zf8vl7gbjgpjn8d30bv83ivdcpxm9m-harfbuzz-11.2.1-dev/include/harfbuzz -I/nix/store/v91r1bkx0ln0yp015m3ij6ab2fz9qslk-glib-2.84.4-dev/include/glib-2.0 -I/nix/store/yhl8qmck6nmll6kcl83xsmw01ymlhwpb-glib-2.84.4/lib/glib-2.0/include -Wall -Wextra -Wno-overloaded-virtual -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-format-security -Wno-sign-compare -fPIC -DOPENSCAD_VERSION=2021.01 -DOPENSCAD_SHORTVERSION=2021.01 -DOPENSCAD_YEAR=2021.0 -DOPENSCAD_MONTH=01.0 -DOPENSCAD_DAY=.0 -DDEBUG -DSTACKSIZE=8388608 -DENABLE_MDI -DUSE_QOPENGLWIDGET -DENABLE_CAIRO -DENABLE_LIBZIP -DLIB3MF_API_2 -DENABLE_LIB3MF -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -DENABLE_OPENCSG -DENABLE_CGAL -DCGAL_HEADER_ONLY -DUSE_SCINTILLA_EDITOR -DQSCINTILLA_DLL -DQT_NO_DEBUG -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_MULTIMEDIA_LIB -DQT_MACEXTRAS_LIB -DQT_GUI_LIB -DQT_CONCURRENT_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/ext/libtess2/Include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtPrintSupport -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtWidgets -I/nix/store/4fgyvg44d1dq3wcf7sndfb2gxh70vbh6-qtmultimedia-5.15.17-dev/include/QtMultimedia -I/nix/store/gkb6zzjqp37n128ap2899965z71g0lr0-qtmacextras-5.15.17-dev/include/QtMacExtras -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtGui -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtConcurrent -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtNetwork -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtCore -Iobjects -I/nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/AGL.framework/Headers -I/nix/store/qd8sscq1h54j54h45fvfqvnnd1gycwqi-eigen-3.4.0-unstable-2022-05-19/include/eigen3 -I/opt/local/include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/mkspecs/macx-clang -o objects/src/Geometry.o src/Geometry.cc
2 warnings generated.
/nix/store/l58v977vq4qy58m1bw60bzrpkywizz2b-clang-wrapper-21.1.1/bin/clang++ -c -pipe -stdlib=libc++ -DSTACKSIZE=8388608 -O2  -arch arm64 -isysroot /nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -mmacosx-version-min=10.9 -I/nix/store/2wsfr9fm6xzy0nrmisxdb01si4d1lyd4-cairo-1.18.4-dev/include/cairo -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include -I/nix/store/7qkcsn97gyn3xnd5jsnrm8mib62pa81c-libzip-1.11.4-dev/../7qkcsn97gyn3xnd5jsnrm8mib62pa81c-libzip-1.11.4-dev/include -I/nix/store/3ydndym6ivg31py7l4flz3lpc28ww15l-libxml2-2.14.6-dev/include/libxml2 -I/nix/store/fl9rvp5x46ahiw9s7inzl5w3bzjjdcfl-lib3mf-2.4.1-dev/include/lib3mf/Bindings/Cpp -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/jxw17p0zgbsl58qn711g6rsf8wgliclz-fontconfig-2.17.1-dev/include -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/w8zf8vl7gbjgpjn8d30bv83ivdcpxm9m-harfbuzz-11.2.1-dev/include/harfbuzz -I/nix/store/v91r1bkx0ln0yp015m3ij6ab2fz9qslk-glib-2.84.4-dev/include/glib-2.0 -I/nix/store/yhl8qmck6nmll6kcl83xsmw01ymlhwpb-glib-2.84.4/lib/glib-2.0/include -Wall -Wextra -Wno-overloaded-virtual -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-format-security -Wno-sign-compare -fPIC -DOPENSCAD_VERSION=2021.01 -DOPENSCAD_SHORTVERSION=2021.01 -DOPENSCAD_YEAR=2021.0 -DOPENSCAD_MONTH=01.0 -DOPENSCAD_DAY=.0 -DDEBUG -DSTACKSIZE=8388608 -DENABLE_MDI -DUSE_QOPENGLWIDGET -DENABLE_CAIRO -DENABLE_LIBZIP -DLIB3MF_API_2 -DENABLE_LIB3MF -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -DENABLE_OPENCSG -DENABLE_CGAL -DCGAL_HEADER_ONLY -DUSE_SCINTILLA_EDITOR -DQSCINTILLA_DLL -DQT_NO_DEBUG -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_MULTIMEDIA_LIB -DQT_MACEXTRAS_LIB -DQT_GUI_LIB -DQT_CONCURRENT_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/ext/libtess2/Include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtPrintSupport -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtWidgets -I/nix/store/4fgyvg44d1dq3wcf7sndfb2gxh70vbh6-qtmultimedia-5.15.17-dev/include/QtMultimedia -I/nix/store/gkb6zzjqp37n128ap2899965z71g0lr0-qtmacextras-5.15.17-dev/include/QtMacExtras -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtGui -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtConcurrent -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtNetwork -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtCore -Iobjects -I/nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/AGL.framework/Headers -I/nix/store/qd8sscq1h54j54h45fvfqvnnd1gycwqi-eigen-3.4.0-unstable-2022-05-19/include/eigen3 -I/opt/local/include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/mkspecs/macx-clang -o objects/src/Polygon2d.o src/Polygon2d.cc
/nix/store/l58v977vq4qy58m1bw60bzrpkywizz2b-clang-wrapper-21.1.1/bin/clang++ -c -pipe -stdlib=libc++ -DSTACKSIZE=8388608 -O2  -arch arm64 -isysroot /nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -mmacosx-version-min=10.9 -I/nix/store/2wsfr9fm6xzy0nrmisxdb01si4d1lyd4-cairo-1.18.4-dev/include/cairo -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include -I/nix/store/7qkcsn97gyn3xnd5jsnrm8mib62pa81c-libzip-1.11.4-dev/../7qkcsn97gyn3xnd5jsnrm8mib62pa81c-libzip-1.11.4-dev/include -I/nix/store/3ydndym6ivg31py7l4flz3lpc28ww15l-libxml2-2.14.6-dev/include/libxml2 -I/nix/store/fl9rvp5x46ahiw9s7inzl5w3bzjjdcfl-lib3mf-2.4.1-dev/include/lib3mf/Bindings/Cpp -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/jxw17p0zgbsl58qn711g6rsf8wgliclz-fontconfig-2.17.1-dev/include -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/w8zf8vl7gbjgpjn8d30bv83ivdcpxm9m-harfbuzz-11.2.1-dev/include/harfbuzz -I/nix/store/v91r1bkx0ln0yp015m3ij6ab2fz9qslk-glib-2.84.4-dev/include/glib-2.0 -I/nix/store/yhl8qmck6nmll6kcl83xsmw01ymlhwpb-glib-2.84.4/lib/glib-2.0/include -Wall -Wextra -Wno-overloaded-virtual -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-format-security -Wno-sign-compare -fPIC -DOPENSCAD_VERSION=2021.01 -DOPENSCAD_SHORTVERSION=2021.01 -DOPENSCAD_YEAR=2021.0 -DOPENSCAD_MONTH=01.0 -DOPENSCAD_DAY=.0 -DDEBUG -DSTACKSIZE=8388608 -DENABLE_MDI -DUSE_QOPENGLWIDGET -DENABLE_CAIRO -DENABLE_LIBZIP -DLIB3MF_API_2 -DENABLE_LIB3MF -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -DENABLE_OPENCSG -DENABLE_CGAL -DCGAL_HEADER_ONLY -DUSE_SCINTILLA_EDITOR -DQSCINTILLA_DLL -DQT_NO_DEBUG -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_MULTIMEDIA_LIB -DQT_MACEXTRAS_LIB -DQT_GUI_LIB -DQT_CONCURRENT_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/ext/libtess2/Include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtPrintSupport -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtWidgets -I/nix/store/4fgyvg44d1dq3wcf7sndfb2gxh70vbh6-qtmultimedia-5.15.17-dev/include/QtMultimedia -I/nix/store/gkb6zzjqp37n128ap2899965z71g0lr0-qtmacextras-5.15.17-dev/include/QtMacExtras -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtGui -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtConcurrent -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtNetwork -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtCore -Iobjects -I/nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/AGL.framework/Headers -I/nix/store/qd8sscq1h54j54h45fvfqvnnd1gycwqi-eigen-3.4.0-unstable-2022-05-19/include/eigen3 -I/opt/local/include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/mkspecs/macx-clang -o objects/src/clipper-utils.o src/clipper-utils.cc
src/node.cc:107:67: error: no member named &#x27;mem_fun&#x27; in namespace &#x27;std&#x27;; did you mean &#x27;mem_fn&#x27;?
  107 |         std::for_each(this-&gt;children.begin(), this-&gt;children.end(), std::mem_fun(&amp;AbstractNode::progress_prepare));
      |                                                                          ^~~~~~~
      |                                                                          mem_fn
/nix/store/zzn2cvbh4gs4zvyv9p52kclqqvqrfb5x-libcxx-19.1.2+apple-sdk-15.5/include/c++/v1/__functional/mem_fn.h:48:81: note: &#x27;mem_fn&#x27; declared here
   48 | inline _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 __mem_fn&lt;_Rp _Tp::*&gt; mem_fn(_Rp _Tp::*__pm) _NOEXCEPT {
      |                                                                                 ^
/nix/store/l58v977vq4qy58m1bw60bzrpkywizz2b-clang-wrapper-21.1.1/bin/clang++ -c -pipe -stdlib=libc++ -DSTACKSIZE=8388608 -O2  -arch arm64 -isysroot /nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -mmacosx-version-min=10.9 -I/nix/store/2wsfr9fm6xzy0nrmisxdb01si4d1lyd4-cairo-1.18.4-dev/include/cairo -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include -I/nix/store/7qkcsn97gyn3xnd5jsnrm8mib62pa81c-libzip-1.11.4-dev/../7qkcsn97gyn3xnd5jsnrm8mib62pa81c-libzip-1.11.4-dev/include -I/nix/store/3ydndym6ivg31py7l4flz3lpc28ww15l-libxml2-2.14.6-dev/include/libxml2 -I/nix/store/fl9rvp5x46ahiw9s7inzl5w3bzjjdcfl-lib3mf-2.4.1-dev/include/lib3mf/Bindings/Cpp -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/jxw17p0zgbsl58qn711g6rsf8wgliclz-fontconfig-2.17.1-dev/include -I/nix/store/khbn41nwaxvkgcjpjcqrpzlisyxzsbbl-freetype-2.13.3-dev/include/freetype2 -I/nix/store/w8zf8vl7gbjgpjn8d30bv83ivdcpxm9m-harfbuzz-11.2.1-dev/include/harfbuzz -I/nix/store/v91r1bkx0ln0yp015m3ij6ab2fz9qslk-glib-2.84.4-dev/include/glib-2.0 -I/nix/store/yhl8qmck6nmll6kcl83xsmw01ymlhwpb-glib-2.84.4/lib/glib-2.0/include -Wall -Wextra -Wno-overloaded-virtual -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-format-security -Wno-sign-compare -fPIC -DOPENSCAD_VERSION=2021.01 -DOPENSCAD_SHORTVERSION=2021.01 -DOPENSCAD_YEAR=2021.0 -DOPENSCAD_MONTH=01.0 -DOPENSCAD_DAY=.0 -DDEBUG -DSTACKSIZE=8388608 -DENABLE_MDI -DUSE_QOPENGLWIDGET -DENABLE_CAIRO -DENABLE_LIBZIP -DLIB3MF_API_2 -DENABLE_LIB3MF -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -DENABLE_OPENCSG -DENABLE_CGAL -DCGAL_HEADER_ONLY -DUSE_SCINTILLA_EDITOR -DQSCINTILLA_DLL -DQT_NO_DEBUG -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_MULTIMEDIA_LIB -DQT_MACEXTRAS_LIB -DQT_GUI_LIB -DQT_CONCURRENT_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/ext/libtess2/Include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtPrintSupport -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtWidgets -I/nix/store/4fgyvg44d1dq3wcf7sndfb2gxh70vbh6-qtmultimedia-5.15.17-dev/include/QtMultimedia -I/nix/store/gkb6zzjqp37n128ap2899965z71g0lr0-qtmacextras-5.15.17-dev/include/QtMacExtras -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtGui -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtConcurrent -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtNetwork -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/include/QtCore -Iobjects -I/nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/nix/store/ghspqwzhdmzj1rss36lcayyc8mrvparn-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/AGL.framework/Headers -I/nix/store/qd8sscq1h54j54h45fvfqvnnd1gycwqi-eigen-3.4.0-unstable-2022-05-19/include/eigen3 -I/opt/local/include -I/nix/store/54a1f96jcq1vdy2alc0whwcdbiimrhxp-qtbase-5.15.17-dev/mkspecs/macx-clang -o objects/src/polyset-utils.o src/polyset-utils.cc
1 error generated.
make: *** [Makefile:2544: objects/src/node.o] Error 1
make: *** Attente des tâches non terminées....</pre>
</details>
</details>

nim65s committed:
openscad: fix on darwin

nim65s committed:
tinygltf: unvendor json & stb

Co-authored-by: jopejoe1 <34899572+jopejoe1@users.noreply.github.com>

nim65s commented:
Wow, somehow lib3mf headers are not installed anymore.
Compared to a previous version where they were installed:
```
- cmake: 4.1.1 -> 4.1.2
- openssl: 3.5.2 -> 3.6.0
- util-linux-minimal: 2.41.1 -> 2.41.2
- python3-minimal: 3.13.7 -> 3.13.8
- libunistring: 1.3 -> 1.4
- coreutils: 9.7 -> 9.8
- go: 1.25.1 -> 1.25.2
- libxml2 2.14.6 -> 2.15.0
- python3.13-pytest: 8.4.1 -> 8.4.2
- python3.13-typing-extensions: 4.14.1 -> 4.14.2
```

Anyways, I guess I'll just fix it.

nim65s committed:
lib3mf: fix header install

Somehow this was working before, but around 2025-10-23, something
changed on master, and nixpkgs-review would stop working on
https://github.com/NixOS/nixpkgs/pull/445078:

```
$ nix build -L github:NixOS/nixpkgs/pull/445078/head#lib3mf.dev
$ eza result-dev/include/
 lib3mf/
$ nix build -L github:NixOS/nixpkgs/pull/445078/merge#lib3mf.dev
$ eza result-dev/include/
"result-dev/include/": No such file or directory (os error 2)
```

The headers are now installed in the build tree:
```
-- Installing: /nix/store/i7qd0hcky65i557fqjl5qq3xrp30hwhz-lib3mf-2.4.1/lib/lib3mf.so.2
-- Installing: /nix/store/i7qd0hcky65i557fqjl5qq3xrp30hwhz-lib3mf-2.4.1/lib/lib3mf.so
-- Installing: /build/source/build/include/lib3mf/Bindings
-- Installing: /build/source/build/include/lib3mf/Bindings/NodeJS
-- Installing: /build/source/build/include/lib3mf/Bindings/NodeJS/lib3mf_nodeaddon.cc
-- Installing: /build/source/build/include/lib3mf/Bindings/NodeJS/lib3mf_nodewrapper.cc
```

Maybe CMake 4.1.1 -> 4.1.2 changed something.

nim65s committed:
openscad-unstable: fix use of our lib3mf cmake export

7c6f434c commented:
Does `openscad-unstable` build for anyone with the fix? For me it fails to find `lib3mf` header `Common/Platform/NMR_WinTypes.h`.

On the other hand, it seems to be already a problem on `master`, so we can defer it to the follow-up

(The changes look reasonable to me and `openscad-unstable` is the only failure in `nixpkgs-review`)

7c6f434c commented:
https://github.com/NixOS/nixpkgs/issues/455261 seems to be related (and on master)

nim65s committed:
lib3mf: fix lib3mf.pc

nim65s commented:
We don't have exactly the same issue, I guess thanks to aa990bec1ec388365dfb08617e031ff35e765694

Still, we can fix the .pc, thanks @adisbladis :)

nim65s commented:
Digging a bit more, I found that debian has the headers. And that is because https://sources.debian.org/src/lib3mf/1.8.1%2Bds-6/CMakeLists.txt#L145

That `install(DIRECTORY Include/Common DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")` was removed in v2: https://github.com/3MFConsortium/lib3mf/pull/178/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL143

But installing directories Common & Model does no fix openscad support for lib3mf v2, because:
```
openscad-unstable> /build/source/src/io/export_3mf_v1.cc:37:10: fatal error: 'lib3mf/Model/COM/NMR_DLLInterfaces.h' file not found
openscad-unstable>    37 | #include <lib3mf/Model/COM/NMR_DLLInterfaces.h>
openscad-unstable>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

And that COM subdirectory was removed in v2, 6y ago: https://github.com/3MFConsortium/lib3mf/tree/v2.0.0-alpha/Include/Model

I guess more work is required to get openscad-unstable working with a lib3mf v2, and maybe that is out of the scope for this PR, because openscad-unstable is broken on master anyways.

7c6f434c commented:
If this is getting merged soon, I guess I can rebase the openscad-unstable fix on top of what is done with lib3mf here.

nim65s commented:
rebased to fix a new conflict in aliases.nix
